src/Analysis.cpp

```diff
@@ -246,6 +234,43 @@ void doAliasReasoning(Instruction *instruction,
         if (CallInst *call = dyn_cast<CallInst>(valueToStore)) {
             ProgramVariable callVar = ProgramVariable(call);
             logout("add alias for analysis storeinst call inst");
+
+            // check for pointer reassignment; if so, the resource becomes un-aliased
+            if (auto pvasRef = programPoint->getPVASRef(receivingVar, false)) {
+                if (pvasRef->containsCallInstVar()) {
+                    logout("pointer reassignment inst " << *instruction);
+                    logout("receiving var " << receivingVar.getRawName());
+
+                    logout(callVar.getRawName()
+                           << " alias to "
+                           << pvasRef->toString(false, true));
+
+
+                    const DebugLoc &debugLoc = call->getDebugLoc();
+
+                    std::string line = rlc_util::getNthLine(optLoadFileName, debugLoc.getLine());
```

Comment:
It worries me that you're using string manipulation for this. What if the C file looks like the following? In general, C programs can contain newlines between any pair of expressions...
```
s.y
=
(char*)malloc(15);
```

In this case, I think this approach won't work. I think instead you should be using the AST to check if the parent of the call instruction is an assignment, and then get the LHS of the assignment from there.

Reply:
(Please add a test like the above, regardless.)